mllib/src/main/scala/org/apache/spark/ml/tree/impl/RandomForest.scala

```diff
@@ -740,7 +740,7 @@ private[ml] object RandomForest extends Logging {
               val categoryStats =
                 binAggregates.getImpurityCalculator(nodeFeatureOffset, featureValue)
               val centroid = if (categoryStats.count != 0) {
-                categoryStats.predict
+                categoryStats.prob(categoryStats.predict)
```

Comment:
Finding the _proportion_ falling in outcome class 1 simply requires division of the counts by a constant. Since we're just using that number for an ordering, constant division won't matter. They are the same. 

My initial comment has a typo. It should say for a "binary **outcome**", not "binary category".